### PR TITLE
add Pdk.bend_points_distance parameter for bends

### PR DESCRIPTION
**Pull request by joamatab**
- add `Pdk.bend_points_distance` parameter for bends
- add database to docs
- make sure euler and arc have always enough points, as suggested by Orion in https://github.com/gdsfactory/gdsfactory/issues/1090

```
npoints=max(npoints, int(360 / dt) + 1)
```

should fix https://github.com/gdsfactory/gdsfactory/issues/1252

@95-martin-orion 
@tvt173

### CODE REVIEW

File: gdsfactory/path.py [hunk 1]
```diff
@@ -1139,7 +1147,11 @@ def euler(
     Rp = R0 / (np.sqrt(p * alpha))
     sp = R0 * np.sqrt(p * alpha)
     s0 = 2 * sp + Rp * alpha * (1 - p)
-    npoints = abs(int(npoints * angle / 360))
+
+    PDK = get_active_pdk()
+    npoints = npoints or abs(int(angle / 360 * radius / PDK.bend_points_distance / 2))
```
Comment (tvt173) [inline gdsfactory/path.py:1152]: is this a good enough approximation for an euler bend? I do this (original xy is the curve with a conservatively high point count)
```python
xy = np.asarray(sg.LineString(xy).simplify(
        tolerance=precision * 0.5 * radius / (radius + cross_section.width + cladding_buffer),
        preserve_topology=False).coords)
```

Comment (tvt173) [reply gdsfactory/path.py:1152]: ah, just noticed the lines below it. nice

Comment (joamatab) [reply gdsfactory/path.py:1152]: we can improve this later, and maybe move it to gdstk.Flexpath

**Comment by joamatab** [comment]
Im not sure why docs are taking so long to merge, i'll debug that in a separate issue

**Pull request merged**